### DISCUSSION TITLE
Creation States and Events — Discussion Open

### DISCUSSION THREAD

**Discussion by modelint** [modelint/shlaer-mellor-metamodel#21] [Creation States and Events]
If you have any feedback or questions on tn1 in the State Metamodel section, feel free to post your thoughts in this discussion area or in a relevant forum of the SM Commons site.

Just hit the green new discussion button in this topic selecting the appropriate category and post your thoughts or simply reply to this post, whichever you prefer.

(Since the SM Commons web/chat services are still being sorted out, this discussion group is provided as an option since I can easily manage it)

– Leon

**Comment by mangoa01** [comment]
Nicely written note that both reviews the OOA96 perspective and justifies the use of asynchronous creation. I would only make one comment. I don't consider the creation event as "special."

Instead I consider the pseudo-initial state as special in the following sense:

1. The pseudo-initial state can only have outbound transitions. With no inbound transitions it makes no sense to have an action associated with the pseudo-initial state, so to prevent any confusion it is forbidden to have an action.
2. Events cannot be ignored from the pseudo-initial state, i.e. there is either a transition according to some event or it is a can't happen. The implication is that you may perform asynchronous creation using multiple events to drive the transition out of the pseudo-initial state and any event used for asynchronous creation which does not have an outbound transition from the pseudo-initial state causes a can't happen error.

From this perspective, all events are the same and can be used to perform transitions between normal states or for the pseudo-initial state. Thus only the pseudo-initial state is special and placing an instance in that state is strictly under the control of the Model Execution domain.

Asynchronous creation is an essential component. In fact I consider it more important than synchronous creation which can be seen as an asynchronous creation followed by an automatic "system devised" synchronization.

**Reply by modelint** [reply]
Thanks and yeah, I overlooked the true destination of the incoming event, the initial psuedo state. Completely agree and now I will have to fix my metamodel accordingly!

I am looking forward to discussing your last point about the system devised synchronization at our next chat.  / Leon

**Comment by michaelmlee-mi** [comment]
Ditto on Andrew's assessment. Per our discussion, I'd suggest putting a summary in the S-M Commons document on this and have it reference this TN. Thanks again for the OOA96/Mellor-Balcer excavation!

**Reply by modelint** [reply]
Done.

**Comment by modelint** [comment]
Have updated the tech note to incorporate Andrew's observation about the initial psuedo-state
